### PR TITLE
python3Packages.xformers: 0.03 -> 0.0.23.post1, add cuda support

### PR DESCRIPTION
**Pull request by CertainLach**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by CertainLach** [comment]
Also something needs to be done with update script, as xformers was downgraded twice at this point:
image
https://github.com/NixOS/nixpkgs/pull/271586#issuecomment-1916656484

**Comment by Tungsten842** [comment]
Could you update this to [v0.0.24](https://github.com/facebookresearch/xformers/releases/tag/v0.0.24)?

**Comment by CertainLach** [comment]
I can, but I won't be able to test it this week.
I have only tested 0.0.23.post1 with vllm on A100.

File: pkgs/development/python-modules/xformers/default.nix [hunk 1]
```diff
@@ -80,6 +99,10 @@ buildPythonPackage {
     transformers
     timm
     # flash-attn
+    # Checks flash-attn version, but fails as no .git found.
+    # Should the check here allow for missing git?
+    # https://github.com/facebookresearch/xformers/blob/e6e66958b29be6ed6428ab0664092e1733d4bdbc/setup.py#L71
+    git
```
Comment (happysalada) [inline pkgs/development/python-modules/xformers/default.nix:105]: you've added the patch to allow explicitely building without git.
So it should just show a git failure log but still succeed without this right ?

Comment (CertainLach) [reply pkgs/development/python-modules/xformers/default.nix:105]: Oh, yes, forgot about that, that's redundant.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 285495` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.xformers</li>
    <li>python311Packages.xformers.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/xformers/default.nix (hunk 1 above)
Comment (happysalada) [reply pkgs/development/python-modules/xformers/default.nix:105]: The nixpkgs review builds no problem. If you remove this, I'm happy to merge this and you can make an update to 0.24 whenever you have time (no rush).

**Comment by happysalada** [comment]
ok let's figure out the update script in a separate step.
Thank you for the contribution!
If you want to do the vllm init, feel free to tag me for it as well.

**Review by dotlambda — COMMENTED** [review]
Please set `passthru.skipBulkUpdate = true` to prevent this from happening again.

**Pull request merged**